### PR TITLE
Reindex

### PR DESCRIPTION
**Pull request by burnes**

### CODE REVIEW

File: src/test/python/reindex.txt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+Via the ``_reindex`` endpoint it is possible to reindex one or all indexes
+with a given query.
+
+Reindex an existing Index
+=========================
+
+    >>> def refresh():
+    ...     post("/_flush")
+    ...     post("/_refresh")
+
```
Comment (dobe) [inline src/test/python/reindex.txt]: put this method into tests.py

**Comment by dobe** [comment]
currently i don't understand why we cannot just change the xcontent of the rest call but override all the transport stuff?

**Comment by burnes** [comment]
Actually the parser for reindex changed, the rest of the code change is more or less refactoring for inheritance.

**Comment by dobe** [comment]
merge master,test, ack

**Pull request merged**